### ISSUE TITLE
Investigate GitHub caching feature

### ISSUE THREAD

**Issue by MrMCake** [Azure/ResourceModules#931]
@renepajta found a feature that seems to allow us to cache e.g. PowerShell modules for performance optimization:

>  "GitHub-hosted runners are started as fresh environments for each job, so if your jobs regularly reuse dependencies, you can consider caching these files to help improve performance. Once the cache is created, it is available to all workflows in the same repository."

Links
- [Advanced workflow features](https://docs.github.com/en/actions/using-workflows/advanced-workflow-features#caching-dependencies)
- [PowerShell Module Cache Marketplace action](https://github.com/marketplace/actions/powershell-module-cache)

**Comment by MariusStorhaug** [comment]
As described in the documentation from the runners, they should have the modules installed. If this is to improve module loading for runners, looking into how to use the already present modules could also speed things up.

GitHub: https://github.com/actions/virtual-environments/blob/ubuntu20/20220123.1/images/linux/Ubuntu2004-Readme.md#az-powershell-modules

ADO: 
https://docs.microsoft.com/en-us/azure/devops/pipelines/agents/hosted?view=azure-devops&tabs=yaml#software

Curious.. the links on ADO point to the same GitHub page though they work differently.

**Comment by MariusStorhaug** [comment]
Looked into the runner, it is storing modules under C:\Modules\az_<version>, or /usr/share/az_<version>

**Comment by MariusStorhaug** [comment]
Tested `Import-Module /usr/share/az_7.1.0/Az` which triggers the import, but fails on the availability of the AzAccount module. Needs just some more investigation I think.

**Comment by MariusStorhaug** [comment]
Also, any reason we are not using the AzurePowerShell built in tasks?

- [azure/powershell@v1 | GitHub](https://github.com/marketplace/actions/azure-powershell-action)
- [Azure PowerShell task | Azure DevOps](https://docs.microsoft.com/en-us/azure/devops/pipelines/tasks/deploy/azure-powershell?view=azure-devops)

**Comment by MrMCake** [comment]
> Also, any reason we are not using the AzurePowerShell built in tasks?
> 
> * [azure/powershell@v1 | GitHub](https://github.com/marketplace/actions/azure-powershell-action)
> * [Azure PowerShell task | Azure DevOps](https://docs.microsoft.com/en-us/azure/devops/pipelines/tasks/deploy/azure-powershell?view=azure-devops)

Nope, and we should. In Azure DevOps we do - and in GitHub we should too. We could approach this once the new dependency pipelines are implemented (#329 ) & side-by-side with #1450
